data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/CidrExpressionFunctionTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@ExtendWith(MockitoExtension.class)
+class CidrExpressionFunctionTest {
+    private CidrExpressionFunction cidrExpressionFunction;
+    private Event testEvent;
+
+    @Mock
+    private Function<Object, Object> testFunction;
+
+    private Event createTestEvent(final Object data) {
+        return JacksonEvent.builder().withEventType("event").withData(data).build();
+    }
+
+    private CidrExpressionFunction createObjectUnderTest() {
+        return new CidrExpressionFunction();
+    }
+
+    @BeforeEach
+    public void setUp() {
+        cidrExpressionFunction = createObjectUnderTest();
+    }
+
+    @Test
+    void testIpv4MatchWithSingleCidrBlock() {
+        String network = "\"192.0.2.0/24\"";
+        testEvent = createTestEvent(Map.of("sourceIp", "192.0.2.3"));
+        Object expressionResult = cidrExpressionFunction.evaluate(List.of("/sourceIp", network), testEvent, testFunction);
+        assertThat((boolean)expressionResult, equalTo(true));
+    }
+
+    @Test
+    void testIpv4MatchWithMultipleCidrBlocks() {
+        String network1 = "\"192.0.2.0/24\"";
+        String network2 = "\"192.168.1.0/24\"";
+        String network3 = "\"10.0.1.0/24\"";
+        testEvent = createTestEvent(Map.of("sourceIp", "192.0.2.3"));
```

Comment:
You should try all the IPs in the range to and one IP outside the range to make sure this is working correctly.

for example, if the CIDR is 192.168.10.0/8, then all 256 entries from 192.168.10.0 to 192.168.10.255 will return true

Reply:
Added parameterized tests for both IPv4 and IPv6 test cases. Tested all addresses in range for the IPv4 case, and tested multiple cases (including lower/upper bounds) for the IPv6 case.